packages/blocks/src/edgeless-text/edgeless-text-block.ts

```diff
@@ -301,10 +300,24 @@ export class EdgelessTextBlockComponent extends EdgelessBlockComponent<
   }
 
   override renderPageContent() {
-    const { color, fontFamily, fontStyle, fontWeight, textAlign } = this.model;
+    const { doc, fontFamily, fontStyle, fontWeight, textAlign } = this.model;
+
+    let color = '#000000';
+    if (doc.awarenessStore.getFlag('enable_color_picker')) {
+      const rootComponent = this
+        .rootComponent as EdgelessRootBlockComponent | null;
+      if (rootComponent) {
+        color = rootComponent.surface.themeObserver.getColor(
```

Comment:
Maybe add a global themeObserver on root service? I think we should try to not to rely on specific block element.